src/cargo/ops/cargo_add/mod.rs

```diff
@@ -331,10 +337,41 @@ fn resolve_dependency(
         };
         selected
     } else if let Some(raw_path) = &arg.path {
+        let base_name_and_value = if let Some(base_name) = &arg.base {
+            let workspace_root = ws.root_manifest().parent().unwrap().to_path_buf();
+            let workspace_root = || Ok(&workspace_root);
+
+            let features = {
+                let empty = Vec::new();
+                let mut warnings = Vec::new();
+                let cargo_features = spec
+                    .manifest()
+                    .original_toml()
+                    .cargo_features
+                    .as_ref()
+                    .unwrap_or(&empty);
+                let features = Features::new(cargo_features, gctx, &mut warnings, true)?;
+                if !warnings.is_empty() {
+                    for warning in warnings {
+                        gctx.shell().warn(warning)?;
+                    }
+                }
+                features
+            };
+            let base_value = lookup_path_base(
+                &PathBaseName::new(base_name.clone())?,
+                gctx,
+                &workspace_root,
+                &features,
+            )?;
+            Some((base_name.clone(), base_value))
+        } else {
+            None
+        };
```

Comment:
If this has switched to `--path` always being local paths (rather than relative to base), could we only store `base` in `PathSource` instead of also a `base_value` and do all of that processing there?

We already have the processing for package-relative paths and this is all very similar.  In fact, each of these reviews is increasing my desire for us to add ``base = "package"`.  `--base package` could then serve as a way to remove a `base`.

Reply:
> do all of that processing there?

I'm not sure what you mean by "there" - I tried moving it into `to_toml` and `update_toml` but we're missing all of the workspace and config information. I could move it into `PathSource::new_with_base`, but I don't think that's really much neater.

Reply:
You are right that passing more parameters along with `crate_root` would be janky.

We have two different means of dealing with relatiev paths.  Bases are handled here and package-relative paths are handled in `to_toml`.  We also have the awkwardness of storing two paths, each used in a different scenario.  I'm trying to find a way to fix both problems.


Let's step back and look at what we are trying to all have work.

We keep paths as absolute because
- Its easier when converting from a `Summary`
- It helps when moving a `Dependency` from a workspace to a package manifest

We could
- Make the path relative only when `base` is present.  This solves part of the problem but not the whole.
- Pass all of those extra parameters to `to_toml`
- Always track a relative path in `PathSource`, requiring us to specially manage `Dependency::from_summary`

Thoughts?

Reply:
I think I've found a reasonable approach.

The key was that the callers of `to_toml` and `from_toml` already have all of the information we need to lookup path bases, they just weren't passing it down to those functions.

The final design is that `PathSource` is always absolute, but now has a `base` field which contains the name of the path base. I'm doing the fixups to set the `base` in `resolve_dependency` and `select_package`, so `Dependency::from(Summary)` stays exactly the same as it currently is.